doesn't compile with Cabal-1.24.2.0

Hi! I'm trying to install it with stack but i'm getting this message:
```
src/Cabal/Haddock.hs:50:12: error:
    Module ‘Distribution.Package’ does not export ‘PackageKey(..)’

src/Cabal/Haddock.hs:80:12: error:
    Module
    ‘Distribution.InstalledPackageInfo’
    does not export
    ‘InstalledPackageInfo_(..)’
```
I'm using:
* stack 1.3 LTS 7.14
* GHC 8.0.1
* Cabal-1.24.2.0
* Haddock Cabal-1.24.2.0


any hint? :)

--- a/src/Cabal/Haddock.hs
+++ b/src/Cabal/Haddock.hs
@@ -47,9 +47,8 @@ module Cabal.Haddock
 -- local
 import Distribution.Package
          ( PackageIdentifier(..)
-         , PackageKey(..)
          , Package(..)
-         , PackageName(..), packageName, PackageId )
+         , PackageName(..), packageName, mkLegacyUnitId, PackageId )
 import qualified Distribution.ModuleName as ModuleName
 import Distribution.PackageDescription as PD
          ( PackageDescription(..), BuildInfo(..), allExtensions
@@ -76,10 +75,8 @@ import Distribution.Simple.BuildPaths ( haddockName,
                                         )
 import Distribution.Simple.PackageIndex (dependencyClosure)
 import qualified Distribution.Simple.PackageIndex as PackageIndex
-import qualified Distribution.InstalledPackageInfo as InstalledPackageInfo
-         ( InstalledPackageInfo_(..) )
 import Distribution.InstalledPackageInfo
-         ( InstalledPackageInfo )
+         ( InstalledPackageInfo, haddockInterfaces, sourcePackageId )
 import Distribution.Simple.Utils
          ( die, copyFileTo, warn, notice, intercalate, setupMessage
          , createDirectoryIfMissingVerbose, withTempFileEx, copyFileVerbose
@@ -195,7 +192,7 @@ haddock pkg_descr lbi suffixes flags computePath = do
             -- Is it possible to get an InstalledPackageInfo here?
             -- If so, it would be better to get the package key from that.
             -- Using OldPackageKey might not work in all cases.
-            , fromFlags (haddockTemplateEnv lbi pkg_id (OldPackageKey pkg_id)) flags
+            , fromFlags (haddockTemplateEnv lbi pkg_id) flags
             , fromPackageDescription pkg_descr ]
 
     let pre c = preprocessComponent pkg_descr c lbi False verbosity suffixes
@@ -543,14 +540,14 @@ haddockPackageFlags lbi clbi computePath = do
     noHaddockWhitelist = map PackageName [ "rts" ]
     interfaceAndHtmlPath :: InstalledPackageInfo -> Maybe (FilePath, FilePath)
     interfaceAndHtmlPath pkg = do
-      interface <- listToMaybe (InstalledPackageInfo.haddockInterfaces pkg)
-      let html = computePath (InstalledPackageInfo.sourcePackageId pkg)
+      interface <- listToMaybe (haddockInterfaces pkg)
+      let html = computePath (sourcePackageId pkg)
       return (interface, html)
 
-haddockTemplateEnv :: LocalBuildInfo -> PackageIdentifier -> PackageKey -> PathTemplateEnv
-haddockTemplateEnv lbi pkg_id pkg_key = (PrefixVar, prefix (installDirTemplates lbi))
-                                : initialPathTemplateEnv pkg_id pkg_key (compilerInfo (compiler lbi))
-                                  (hostPlatform lbi)
+haddockTemplateEnv :: LocalBuildInfo -> PackageIdentifier -> PathTemplateEnv
+haddockTemplateEnv lbi pkg_id = (PrefixVar, prefix (installDirTemplates lbi))
+                         : initialPathTemplateEnv pkg_id (mkLegacyUnitId pkg_id) (compilerInfo (compiler lbi))
+                           (hostPlatform lbi)
 
 -- --------------------------------------------------------------------------
 -- hscolour support

--- a/src/Cabal/Haddock.hs
+++ b/src/Cabal/Haddock.hs
@@ -86,6 +86,7 @@ import Distribution.Simple.Utils
          , withTempDirectoryEx, matchFileGlob
          , findFileWithExtension, findFile
          , TempFileOptions(..), defaultTempFileOptions )
+import Distribution.System (buildPlatform)
 import Distribution.Text
          ( display, simpleParse )
 import Distribution.Utils.NubList
@@ -487,7 +488,7 @@ renderPureArgs version comp args = concat
               . fromFlag . argTitle $ args,
      [ "--optghc=" ++ opt | isVersion2
                           , (opts, _ghcVer) <- flagToList (argGhcOptions args)
-                          , opt <- renderGhcOptions comp opts ],
+                          , opt <- renderGhcOptions comp buildPlatform opts ],
      maybe [] (\l -> ["-B"++l]) $ guard isVersion2 >> flagToMaybe (argGhcLibDir args), -- error if isVersion2 and Nothing?
      argTargets $ args
     ]

--- a/src/Main.hs
+++ b/src/Main.hs
@@ -1,10 +1,8 @@
 {-# LANGUAGE RecordWildCards #-}
 {-# OPTIONS -fno-warn-name-shadowing #-}
 import Options.Applicative
-import Options.Applicative.Types
 import Cabal.Simple
 import Cabal.Haddock
-import Control.Monad hiding (forM_)
 import Data.Monoid
 import qualified Data.Set as Set
 import Text.Printf
@@ -45,8 +43,6 @@ optParser =
         )  
     <*> strOption (short 'o' <> metavar "OUTPUT-PATH" <> help "Directory where html files will be placed")
     <*> many (argument str (metavar "PACKAGE-PATH"))
-  where
-    readEither s = case reads s of [(n, "")] -> Just n; _ -> Nothing
 
 getPackageNames
   :: Verbosity

--- a/standalone-haddock.cabal
+++ b/standalone-haddock.cabal
@@ -24,7 +24,7 @@ executable standalone-haddock
   -- other-extensions:    
   build-depends:
     base == 4.*,
-    Cabal >= 1.22,
+    Cabal >= 1.24,
     filepath,
     directory,
     containers,

